bin/clover/src/pipelines/gcp/utils.ts

```diff
@@ -1,50 +1,78 @@
-    const hasUpdateablePattern = /\b(and|or) (be )?(updated|changed|modified)\b/i.test(description);
-    const isCreateOnly = hasCreateOnlyPattern && !hasUpdateablePattern;
+/**
+ * URL path segments that indicate a location (extract just the name).
+ * When these appear as the second-to-last segment, we extract just the final name.
+ */
+const LOCATION_SEGMENTS = new Set(["regions", "zones", "locations"]);
 
-    if (isCreateOnly) {
-      createOnlyProps.push(propName);
+/**
+ * Recursively normalizes all GCP URLs in a resource payload.
+ *
+ * Normalization rules:
+ * - Location URLs (ending with /regions/{name}, /zones/{name}, /locations/{name})
+ *   → extract just the name
+ * - Project-scoped URLs → extract full path from "projects" onwards
+ * - Non-project URLs (e.g., Storage) → extract path after API version (v1, v2, etc.)
+ *
+ * @param obj - The object to normalize (can be nested)
+ * @returns A new object with all GCP URLs converted to appropriate formats
+ *
+ * @example
+ * // Location URL
+ * "https://compute.googleapis.com/compute/v1/projects/p/regions/us-central1"
+ * // → "us-central1"
+ *
+ * // Project-scoped resource URL
+ * "https://compute.googleapis.com/compute/v1/projects/p/global/networks/my-network"
+ * // → "projects/p/global/networks/my-network"
+ *
+ * // Non-project URL (Storage)
+ * "https://storage.googleapis.com/storage/v1/b/my-bucket/o/my-object"
+ * // → "b/my-bucket/o/my-object"
+ */
+export function normalizeGcpResourceValues<T>(obj: T): T {
+  if (obj === null || obj === undefined) return obj;
+  if (Array.isArray(obj)) return obj.map(item => normalizeGcpResourceValues(item)) as T;
+  if (typeof obj === "object") {
+    const normalized: Record<string, unknown> = {};
+    for (const [key, value] of Object.entries(obj)) {
+      if (typeof value === "string" && GCP_URL_PATTERN.test(value)) {
+        const pathParts = new URL(value).pathname.split("/").filter(Boolean);
+        if (pathParts.length >= 2 && LOCATION_SEGMENTS.has(pathParts[pathParts.length - 2])) {
+          normalized[key] = pathParts[pathParts.length - 1];
+        } else {
+          const projectsIdx = pathParts.indexOf("projects");
+          if (projectsIdx !== -1) {
+            normalized[key] = pathParts.slice(projectsIdx).join("/");
+          } else {
+            // For non-project APIs (e.g., Storage), extract after API version (v1, v2, etc.)
+            const versionIdx = pathParts.findIndex(p => /^v\d+/.test(p));
+            normalized[key] = versionIdx !== -1 && versionIdx + 1 < pathParts.length
+              ? pathParts.slice(versionIdx + 1).join("/")
+              : pathParts[pathParts.length - 1] || value;
+          }
+        }
```

Comment:
I think I understand what this does, but if it breaks, I don't know that I could fix it :laughing: